ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/LagReportingAgent.java

```diff
@@ -171,12 +167,11 @@ public void receiveHostLag(final LagReportingMessage lagReportingMessage) {
   public Optional<LagInfoEntity> getHostsPartitionLagInfo(
       final KsqlHost host, final QueryStateStoreId queryStateStoreId, final int partition) {
     final Set<KsqlHost> aliveHosts = aliveHostsRef.get();
-    final LagInfoEntity lagInfo = receivedLagInfo
-        .getOrDefault(host, EMPTY_HOST_STORE_LAGS)
-        .getStateStoreLagsOrDefault(queryStateStoreId, EMPTY_STATE_STORE_LAGS)
-        .getLagByPartition(partition);
-    if (aliveHosts.contains(host) && lagInfo != null) {
-      return Optional.of(lagInfo);
+    final Optional<LagInfoEntity> lagInfo = Optional.ofNullable(receivedLagInfo.get(host))
+        .flatMap(hostStoreLags -> hostStoreLags.getStateStoreLags(queryStateStoreId))
+        .flatMap(stateStoreLags -> stateStoreLags.getLagByPartition(partition));
+    if (aliveHosts.contains(host)) {
```

Comment:
nit: if you invert this `if` you could move it before the construction of the `lagInfo` var.

e.g.

```java
  if (!aliveHosts.contains(host)) {
     return Optional.empty();
   }

   final Optional<LagInfoEntity> lagInfo = ...
   return lagInfo;
```

Reply:
Done